changes/12550-add-validation-for-policy-specs

```diff
@@ -0,0 +1 @@
+- add validation when running `fleetctl apply` on policies.  Fixes issue 12550
```

Comment:
AFAIK we don't add issue numbers here.

Maybe mention briefly what type of validation was added.

Reply:
updated